root/static/scripts/common/MB/Control/Menu.js

```diff
@@ -41,6 +41,10 @@ MB.Control.HeaderMenu = function () {
         }, self.timeout_msecs);
     });
 
+    $('.header .menu-header').click(function () {
+        $(this).siblings('ul').toggle();
```

Comment:
Works fine. Honestly I'd like to get rid of hover completely, but small steps and all that :p